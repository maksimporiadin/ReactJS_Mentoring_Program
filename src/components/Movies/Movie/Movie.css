.movie-container {
    /* //height: 300px; */
    display: flex;
    padding: 30px;
    width: 100%;
}

@media all and (min-width: 40em) {
    .movie-container {
        width: 50%;
    }
}

@media all and (min-width: 60em) {
    .movie-container {
        width: 33.33%;
    }
}

@media all and (min-width: 100em) {
    .movie-container {
        width: 20%;
    }
}

.movie-content {
    display: flex;
    flex-direction: column;
    width: 260px;
}

.movie-poster {
    vertical-align: middle;
    border-style: none;
    width: 260px;
    min-height: 378px;
    align-self: center;
    margin-bottom: 10px;
}

.movie-description-content {
    display: grid;
    grid-template-columns: 210px 50px;
}

.movie-description-gentes {
    grid-column: 1 / 3;
}

.movie-description-date {
    justify-self: center;
}

.movie-description-date span {
    padding: 2px 10px;
    border: 1px solid #cccccc;
}