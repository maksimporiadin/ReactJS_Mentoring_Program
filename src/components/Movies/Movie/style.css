.movie-container {
    display: flex;
    justify-content: center;
    padding: 30px;
    width: 100%;
}

@media all and (min-width: 40em) {
    .movie-container {
        width: 50%;
    }
}

@media all and (min-width: 60em) {
    .movie-container {
        width: 33.33%;
    }
}

@media all and (min-width: 100em) {
    .movie-container {
        width: 20%;
    }
}

.movie-content {
    display: flex;
    flex-direction: column;
    width: 260px;
}

.movie-poster {
    vertical-align: middle;
    border-style: none;
    width: 260px;
    min-height: 378px;
    align-self: center;
    margin-bottom: 10px;
}

.movie-description-content {
    display: grid;
    grid-template-columns: 210px 50px;
}

.movie-description-gentes {
    grid-column: 1 / 3;
}

.movie-description-date {
    justify-self: center;
}

.movie-description-date span {
    padding: 0px 7px;
    border: 1px solid #cccccc;
}