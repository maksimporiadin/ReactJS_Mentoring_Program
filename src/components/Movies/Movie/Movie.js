import React from 'react';
import { Link } from 'react-router-dom'

import './Movie.css';

const Movie = ({ movie }) => (
    <div className="movie-container">
        <div className="movie-content">
            <img className="movie-poster"
                 src={movie.poster_path}
                 alt={movie.title} />
            <div className="movie-description-content">
                <div className="movie-description-link">
                    <Link to={`/movies/${movie.id}`}>{movie.title}</Link>
                </div>
                <div className="movie-description-date">
                    <span>{movie.release_date.slice(0, 4)}</span>
                </div>
                <div className="movie-description-gentes">{`${movie.genres.join(' & ')} Movie`}</div>
            </div>
        </div>
    </div>
);

export default Movie;