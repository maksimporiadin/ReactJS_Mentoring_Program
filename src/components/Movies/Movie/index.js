// @flow

import React from 'react';
import { Link } from 'react-router-dom';

import './style.css';

type MovieProps = {
    movie: Object;
}

const Movie = (props: MovieProps) => {
    const { movie } = props;

    return (
    <div className="movie-container">
        <div className="movie-content">
            <img className="movie-poster"
                 src={movie.poster_path}
                 alt={movie.title} />
            <div className="movie-description-content">
                <div className="movie-description-link">
                    <Link to={`/movie/${movie.id}`}>{movie.title}</Link>
                </div>
                <div className="movie-description-date">
                    <span>{movie.release_date.slice(0, 4)}</span>
                </div>
                <div className="movie-description-gentes">{`${movie.genres.join(' & ')} Movie`}</div>
            </div>
        </div>
    </div>
)};

export default Movie;