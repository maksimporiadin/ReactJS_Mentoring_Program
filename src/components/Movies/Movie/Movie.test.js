import 'jsdom-global/register';
import React from 'react';

import Movie from './index';

import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});

describe('Movie', () => {
    let wrapper;
    const defaultProps = {
        movie: {
            poster_path: 'https://movie.url',
            title: 'title',
            genres: ['genre1', 'genre2'],
            release_date: '2014_12_12',
            id: 124,
        }
    };

    beforeEach(()=>{
        jest.resetAllMocks();
        wrapper = shallow(<Movie {...defaultProps} />)
    });

    it('is renders', () => {
        expect(wrapper).toMatchSnapshot();
    });
});