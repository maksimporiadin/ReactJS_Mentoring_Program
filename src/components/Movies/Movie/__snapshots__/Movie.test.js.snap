// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie is renders 1`] = `
<div
  className="movie-container"
>
  <div
    className="movie-content"
  >
    <img
      alt="title"
      className="movie-poster"
      src="https://movie.url"
    />
    <div
      className="movie-description-content"
    >
      <div
        className="movie-description-link"
      >
        <Link
          replace={false}
          to="/movies/124"
        >
          title
        </Link>
      </div>
      <div
        className="movie-description-date"
      >
        <span>
          2014
        </span>
      </div>
      <div
        className="movie-description-gentes"
      >
        genre1 & genre2 Movie
      </div>
    </div>
  </div>
</div>
`;
